### ISSUE TITLE
If subtask in concurrent mode fails, the renderer gets stuck and fails silently

### ISSUE THREAD

**Issue by IgnusG** [cenk1cenk2/listr2#4]
If a subtask throws an error in concurrent mode the renderer stops animating the rest of the subtasks. No error is printed and listr2 fails silently. Not sure if I have to use `manager.indent` for this to work.

```javascript
const manager = new Manager();

manager.add({
  title: 'Do stuff',
  task: async () => {
    return new Listr([
      { title: 'Task 1', task: async () => { throw new Error('FAIL') } },
      { title: 'Task 2', task: async () => { await delay(1000); return true; } }
    ], { concurrent: true })
  }
});

manager.runAll().catch(() => null);
```

![listr2-bug](https://user-images.githubusercontent.com/6438760/77911862-d229ca80-7291-11ea-99b4-8e023d6ce46f.gif)

**Comment by cenk1cenk2** [comment]
Hmm. Manager indent is just a jumper function that does the same thing without writing to whole thingy, which made things more cleaner for me. So I think they should have the same effect.

I will have a look this evening for this. But I think this issue can also occur from not catching the promise properly.
```typescript
let ctx: SomeCtxType
try {
  ctx = await manager.runAll()
} catch (e) {
  console.error(e)
}
```

**Comment by IgnusG** [comment]
I do have the catch at the end in my code. Sorry skipped it for brevity purposes 😅

**Comment by cenk1cenk2** [comment]
@IgnusG Merged and publish with #7. I am having a look at the other issue #5 to inject custom functions. There will be no type checking in my solution but I am thinking something like the one below:

```typescript
  manager.add([
    {
      title: 'Something',
      task: (ctx, task): void => {
        task.extend('logThis')('string')
      }
    }
  ])

  try {
    ctx = await manager.runAll({ extend: { logThis: (string): void => { console.log(string) } } })
  // eslint-disable-next-line no-empty
  } catch (e){

  }
```

**Comment by cenk1cenk2** [comment]
Will this work for your application?
It will now clean up the render. And mark the rest of the tasks with red. I made my effort to make only the subtasks red instead of the sibling tasks, but I have to change the whole structure to do that, which seems unnecessary.

## Case 1
```typescript
manager.add([
  {
    title: 'Do stuff',
    task: async () => {
      return new Listr([
        { title: 'Task 1', task: async () => { throw new Error('FAIL') } },
        { title: 'Task 2', task: async (ctx, task) => { await delay(1000); task.title = 'I succeed'; return true } }
      ], {
        concurrent: true,
        exitOnError: false
      })
    }
  },
  ...someMoreTasks
], { exitOnError: true })

```
### Result
![image](https://user-images.githubusercontent.com/26881592/77930670-7835fe80-72ab-11ea-85bb-546eda86258b.png)

## Case 2:
```typescript
manager.add([
  {
    title: 'Do stuff',
    task: async () => {
      return new Listr([
        { title: 'Task 1', task: async () => { throw new Error('FAIL') } },
        { title: 'Task 2', task: async (ctx, task) => { await delay(1000); task.title = 'I succeed'; return true } }
      ], {
        concurrent: true,
        exitOnError: true
      })
    }
  },
  ...someMoreTasks
], { exitOnError: true })
```
### Result
![image](https://user-images.githubusercontent.com/26881592/77930545-4fae0480-72ab-11ea-8360-6ecbc3976252.png)

## As a more wider view in example.ts
![image](https://user-images.githubusercontent.com/26881592/77934813-e9c47b80-72b0-11ea-924f-3e7ca8366e4e.png)
